profiling/src/internal/observation/observations.rs

```diff
@@ -160,50 +161,15 @@ mod test {
         o.add(s1, None, vec![1, 2, 3]);
         o.add(s1, None, vec![4, 5, 6]);
         o.add(s2, None, vec![7, 8, 9]);
-        o.iter().for_each(|(k, ts, v)| {
-            assert!(ts.is_none());
-            if k == s1 {
-                assert_eq!(v, vec![5, 7, 9]);
-            } else if k == s2 {
-                assert_eq!(v, vec![7, 8, 9]);
-            } else {
-                panic!("Unexpected key");
-            }
-        });
-        // Iter twice to make sure there are no issues doing that
-        o.iter().for_each(|(k, ts, v)| {
-            assert!(ts.is_none());
-            if k == s1 {
-                assert_eq!(v, vec![5, 7, 9]);
-            } else if k == s2 {
-                assert_eq!(v, vec![7, 8, 9]);
-            } else {
-                panic!("Unexpected key");
-            }
-        });
         o.add(s3, t1, vec![10, 11, 12]);
```

Comment:
We're losing some visibility here.  Maybe that's ok, but worth calling out

Reply:
Actually, not really :wink: . The existing tests were checking the following, in three separate passes:

```rust
        o.add(s1, None, vec![1, 2, 3]);
        o.add(s1, None, vec![4, 5, 6]);
        o.add(s2, None, vec![7, 8, 9]);
        o.iter().for_each(|(k, ts, v)| {
            assert!(ts.is_none());
            if k == s1 {
                assert_eq!(v, vec![5, 7, 9]);
            } else if k == s2 {
                assert_eq!(v, vec![7, 8, 9]);
            } else {
                panic!("Unexpected key");
            }
```

* That the two samples for s1 were aggregated
* That the sample for s2 was kept separately

```rust
        o.add(s3, t1, vec![10, 11, 12]);
        o.iter().for_each(|(k, ts, v)| {
            if k == s1 {
                assert_eq!(v, vec![5, 7, 9]);
                assert!(ts.is_none());
            } else if k == s2 {
                assert_eq!(v, vec![7, 8, 9]);
                assert!(ts.is_none());
            } else if k == s3 {
                assert_eq!(v, vec![10, 11, 12]);
                assert_eq!(ts, t1);
            } else {
                panic!("Unexpected key");
            }
        });
```

* That adding timeline to the mix didn't change things

```rust
        o.add(s2, t2, vec![13, 14, 15]);
```

* That having two samples with the same stack, one with timeline and another with timeline would not be aggregated together

...and I argue that all 4 above are still being covered in the last check. I even called out what we're testing with comments :smile: 

```rust
        o.into_iter().for_each(|(k, ts, v)| {
            if k == s1 {
                // Observations without timestamp, these are aggregated together
                assert_eq!(v, vec![5, 7, 9]);
                assert!(ts.is_none());
            } else if k == s2 {
                // Same stack with and without timestamp
                if ts.is_some() {
                    assert_eq!(v, vec![13, 14, 15]);
                    assert_eq!(ts, t2);
                } else {
                    assert_eq!(v, vec![7, 8, 9]);
                    assert!(ts.is_none());
                }
            } else if k == s3 {
                // Observation with timestamp
                assert_eq!(v, vec![10, 11, 12]);
                assert_eq!(ts, t1);
            } else {
                panic!("Unexpected key");
            }
        });
```

The remaining thing that was tested was that you could iterate multiple times, but on purpose we want the `into_iter` to release memory (right...?) so that behavior was actually removed.